pysat/_instrument.py

```diff
@@ -2965,6 +2965,13 @@ def load(self, yr=None, doy=None, end_yr=None, end_doy=None, date=None,
         else:
             use_header = True
 
+        # Provide user friendly error if there is no data
+        if len(self.files.files) == 0:
+            estr = ''.join(('No files found for Instrument. Please confirm ',
+                            'that data is present on the system and that ',
+                            "pysat.params['data_dirs'] is set correctly."))
+            raise OSError(estr)
```

Comment:
Not sure an error is the right tack to take here.  At the moment, I am very sure people have code that will break for this, because they have set up behaviour to check for empty after loading data and take steps from there.  I'm not opposed to this change in behaviour later on, but not for this release.

Reply:
The code leads to an error without intentionally raising this user friendly error, as per #1178. Though that error is an IndexError. Could be a warning for now with a heads up things will change in the future.